### PR TITLE
Add prune progress display

### PR DESCRIPTION
**Pull request by Abogical**
For #1721

### CODE REVIEW

File: src/borg/archiver.py [hunk 1]
```diff
@@ -1016,18 +1016,26 @@ def do_prune(self, args, repository, manifest, key):
...
                     else:
-                        if args.output_list:
-                            list_logger.info('Pruning archive: %s' % format_archive(archive))
-                        Archive(repository, key, manifest, archive.name, cache).delete(stats, forced=args.forced)
+                        archives_to_delete.append(archive)
                 else:
                     if args.output_list:
                         list_logger.info('Keeping archive: %s' % format_archive(archive))
+            if args.progress:
+                len_archives_to_delete = len(archives_to_delete)
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py]: isn't this the same as len(to_delete)?

if so, can you stay close to the old code to do what you want?

Comment (Abogical) [inline src/borg/archiver.py]: Ok that change was dumb. I though maybe not all of the `to_delete` set was in `archive_checkpoints`. Sorry about that, will change it.

File: src/borg/archiver.py [hunk 2]
```diff
@@ -1024,7 +1028,11 @@ def do_prune(self, args, repository, manifest, key):
                     else:
                         if args.output_list:
                             list_logger.info('Pruning archive: %s' % format_archive(archive))
-                        Archive(repository, key, manifest, archive.name, cache).delete(stats, forced=args.forced)
+                        Archive(repository, key, manifest, archive.name, cache,
+                                progress=args.progress).delete(stats, forced=args.forced)
+                        if args.progress:
+                            archives_deleted += 1
+                            logger.warning('(%d/%d) archives pruned' % (archives_deleted, to_delete_len))
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py]: this is progress display (information), not a warning.

why didn't you use the ProgressIndicator used elsewhere?

Comment (Abogical) [inline src/borg/archiver.py]: Will change warning to info...

File: src/borg/archiver.py [hunk 3]
```diff
@@ -1024,7 +1028,11 @@ def do_prune(self, args, repository, manifest, key):
                     else:
                         if args.output_list:
                             list_logger.info('Pruning archive: %s' % format_archive(archive))
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py]: see this ^^^

File: src/borg/archiver.py [hunk 4]
```diff
@@ -1024,7 +1028,11 @@ def do_prune(self, args, repository, manifest, key):
...
-                        Archive(repository, key, manifest, archive.name, cache).delete(stats, forced=args.forced)
+                        Archive(repository, key, manifest, archive.name, cache,
+                                progress=args.progress).delete(stats, forced=args.forced)
+                        if args.progress:
+                            archives_deleted += 1
+                            logger.info('(%d/%d) archives pruned' % (archives_deleted, to_delete_len))
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py]: ... and this ^^^.

So, the only difference is that one is output before and the other after deletion of one archive - and the counter / total.

I think one log line is enough for that (and it maybe should be before deleting the archive).

Check how 1.0-maint handles this and also how it is handled at other places (--list vs. --progress).

Also, the progress output happens on the borg.output.progress logger elsewhere.

Comment (Abogical) [inline src/borg/archiver.py]: If i understand correctly, Is this commit useless?

Comment (ThomasWaldmann) [inline src/borg/archiver.py]: well, it does not list counter/total info (but it could).

Comment (Abogical) [inline src/borg/archiver.py]: Good idea, will do that.

**Comment by Abogical** [comment]
I'm not sure if you are waiting for another commit, but I'm done here.

**Comment by Abogical** [comment]
Accidentally closed PR, sorry about that.

Not sure if you're waiting for another commit, but I'm done here.

**Comment by enkore** [comment]
Thanks!

**Comment by Abogical** [comment]
Your welcome, I'm waiting for feedback on #1721.

**Pull request merged**